lib/query-interface.js

```diff
@@ -495,18 +495,20 @@ QueryInterface.prototype.insert = function(instance, tableName, values, options)
   });
 };
 
-QueryInterface.prototype.upsert = function(tableName, values, updateValues, model, options) {
+QueryInterface.prototype.upsert = function(tableName, valuesByField, updateValues, model, options) {
```

Comment:
Right but the end user would still pass the values to upsert via attribute names.
However we already do mapping here: https://github.com/sequelize/sequelize/blob/master/lib/model.js#L2032 so not quite sure why the fix is needed in the first place.